src/main/java/datastax/astra/migrate/OriginCountJobSession.java

```diff
@@ -0,0 +1,157 @@
+        sourceSelectStatement = sourceSession.prepare(
+                "select " + selectCols + " from " + sourceKeyspaceTable + " where token(" + partionKey.trim()
+                        + ") >= ? and token(" + partionKey.trim() + ") <= ?  " + sourceSelectCondition + " ALLOW FILTERING");
+
+    }
+
+    public void getData(BigInteger min, BigInteger max) {
+        logger.info("TreadID: " + Thread.currentThread().getId() + " Processing min: " + min + " max:" + max);
+        int maxAttempts = maxRetries;
+        for (int retryCount = 1; retryCount <= maxAttempts; retryCount++) {
+
+            try {
+                ResultSet resultSet = sourceSession.execute(sourceSelectStatement.bind(hasRandomPartitioner ? min : min.longValueExact(), hasRandomPartitioner ? max : max.longValueExact()));
+                Collection<CompletionStage<AsyncResultSet>> writeResults = new ArrayList<CompletionStage<AsyncResultSet>>();
+
+                // cannot do batching if the writeFilter is greater than 0 or
+                // maxWriteTimeStampFilter is less than max long
+                // do not batch for counters as it adds latency & increases chance of discrepancy
+                if (batchSize == 1 || writeTimeStampFilter || isCounterTable) {
+                    for (Row sourceRow : resultSet) {
+                        readLimiter.acquire(1);
+
+                        if(checkTableforColSize) {
+                            int rowColcnt = GetRowColumnLength(sourceRow, filterColType, filterColIndex);
+                            String result = "";
+                            if (rowColcnt > 1024 * 1024 * 10) {
+                                for (int index = 0; index < checkTableforColSizeTypes.size(); index++) {
+                                    MigrateDataType dataType = checkTableforColSizeTypes.get(index);
+                                    Object colData = getData(dataType, index, sourceRow);
+                                    String[] colName = checkTableforselectCols.split(",");
+                                    result = result + " - " + colName[index] + " : " + colData;
+                                }
+                                logger.error("ThreadID: " + Thread.currentThread().getId() + result + " - " + filterColName + " length: " + rowColcnt);
+                                continue;
+                            }
+                        }
+                    }
+
+                } else {
+                    BatchStatement batchStatement = BatchStatement.newInstance(BatchType.UNLOGGED);
+                    for (Row sourceRow : resultSet) {
+                        readLimiter.acquire(1);
+                        writeLimiter.acquire(1);
+
+                        if(checkTableforColSize) {
+                            int rowColcnt = GetRowColumnLength(sourceRow, filterColType, filterColIndex);
+                            String result = "";
+                            if (rowColcnt > 1024 * 1024 * 10) {
```

Comment:
Can this variable be loaded from config 
`int fieldGuardraillimitMB = load from config e.g. 10;
`
Also can this variable name be improved to show whats happening?
`int sizeInMB = fieldGuardraillimitMB * org.apache.commons.io.FileUtils.ONE_MB;`(https://commons.apache.org/proper/commons-io/javadocs/api-2.2/org/apache/commons/io/FileUtils.html#ONE_MB)